zellij-server/src/os_input_output.rs

```diff
@@ -336,6 +398,20 @@ impl ServerOsApi for ServerOsInputOutput {
     fn load_palette(&self) -> Palette {
         default_palette()
     }
+    #[cfg(target_os = "macos")]
+    fn get_cwd(&self, pid: RawFd) -> Option<PathBuf> {
+        match darwin_libproc::pid_cwd(pid) {
+            Ok(cwd) => Some(cwd),
+            Err(_) => None,
+        }
+    }
+    #[cfg(target_os = "linux")]
```

Comment:
I resolved this by just returning None for non Linux/MacOS systems. I know very little about alternative Unixes so I don't really feel confident doing anything else here. I did quickly google whether or not FreeBSD has a /proc directory and apparently it does but you have to mount it first. Given that, it's probably not trivial to support everything.

Reply:
That's a good solution, but correct me if I'm wrong - right now in those systems it won't compile though right? Because there is no specific function with the correct "is unix" conditional compilation? Or am I missing something?

Reply:
Right, I resolved that here https://github.com/zellij-org/zellij/pull/691/commits/5f13c9520cc828203ebe9aa23c1a04c55781017b by using:
```rust
#[cfg(all(not(target_os = "linux"), not(target_os = "macos")))]
```
Which is admittedly a bit confusing. I couldn't just use `#[cfg(target_family = "unix")]` because Linux is considered a Unix by Rust, and it'll throw a duplicate definition error. The boolean logic there matches everything that's not either Linux or MacOS, which includes the larger family of Unix OS's Rust supports as well as Windows, Android, ..etc.

Reply:
No no, that's good - I missed it before, sorry!